src/infra/flow-system/postgres/src/flow_system_event_store_postgres.rs

```diff
@@ -0,0 +1,257 @@
+    async fn get_events(
+        &self,
+        flow_key: &FlowKey,
+        opts: GetEventsOpts,
+    ) -> EventStream<FlowConfigurationEvent> {
+        let maybe_from_id = opts.from.map(EventID::into_inner);
+        let maybe_to_id = opts.to.map(EventID::into_inner);
+
+        match flow_key.clone() {
+            FlowKey::Dataset(fk_dataset) => {
+                self.get_dataset_events(fk_dataset, maybe_from_id, maybe_to_id)
+                    .await
+            }
+            FlowKey::System(fk_system) => {
+                self.get_system_events(fk_system, maybe_from_id, maybe_to_id)
+                    .await
+            }
+        }
+    }
+
+    async fn save_events(
+        &self,
+        flow_key: &FlowKey,
+        events: Vec<FlowConfigurationEvent>,
+    ) -> Result<EventID, SaveEventsError> {
+        let mut tr = self.transaction.lock().await;
+        let connection_mut = tr.connection_mut().await?;
+
+        let mut query_builder = match flow_key {
+            FlowKey::Dataset(fk_dataset) => {
+                let mut query_builder = QueryBuilder::<Postgres>::new(
+                    r#"
+INSERT INTO dataset_flow_configuration_events (dataset_id, dataset_flow_type, event_type, event_time, event_payload)
+"#,
+                );
+
+                query_builder.push_values(events, |mut b, event| {
+                    b.push_bind(fk_dataset.dataset_id.to_string());
+                    b.push_bind(fk_dataset.flow_type);
+                    b.push_bind(event.typename());
+                    b.push_bind(event.event_time());
+                    b.push_bind(serde_json::to_value(event).unwrap());
+                });
+
+                query_builder
+            }
+            FlowKey::System(fk_system) => {
+                let mut query_builder = QueryBuilder::<Postgres>::new(
+                    r#"
+INSERT INTO system_flow_configuration_events (system_flow_type, event_type, event_time, event_payload)
+"#,
+                );
+
+                query_builder.push_values(events, |mut b, event| {
+                    b.push_bind(fk_system.flow_type);
+                    b.push_bind(event.typename());
+                    b.push_bind(event.event_time());
+                    b.push_bind(serde_json::to_value(event).unwrap());
+                });
+
+                query_builder
+            }
+        };
+
+        query_builder.push("RETURNING event_id");
+
+        #[derive(FromRow)]
+        struct ResultRow {
+            event_id: i64,
+        }
+
+        let rows = query_builder
+            .build_query_as::<ResultRow>()
+            .fetch_all(connection_mut)
+            .await
+            .int_err()?;
+
+        if let Some(last_row) = rows.last() {
+            Ok(EventID::new(last_row.event_id))
+        } else {
+            let event_id = i64::try_from(self.len().await.int_err()?).int_err()?;
+
+            Ok(EventID::new(event_id))
+        }
```

Comment:
I don't remember why I wrote this condition in tasks doain, and I don't understand what it is doing here as wel.
In partcular, I think INSERT will always return a good event id.